server/sonar-ce/src/test/java/org/sonar/ce/cluster/HazelcastClientWrapperTest.java

```diff
@@ -0,0 +1,137 @@
+import java.util.Set;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.sonar.api.config.MapSettings;
+import org.sonar.api.config.PropertyDefinitions;
+import org.sonar.api.config.Settings;
+import org.sonar.process.NetworkUtils;
+import org.sonar.process.ProcessProperties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.MapEntry.entry;
+
+public class HazelcastClientWrapperTest {
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public TestRule safeGuard = new DisableOnDebug(Timeout.seconds(10));
+
+  @Test
+  public void missing_CLUSTER_ENABLED_throw_ISE() {
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("Cluster is not enabled");
+
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_ENABLED);
+    HazelcastClientWrapper hazelcastClientWrapper = new HazelcastClientWrapper(settings);
+    hazelcastClientWrapper.stop();
+  }
+
+  @Test
+  public void missing_CLUSTER_NAME_throw_ISE() {
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("sonar.cluster.name is missing");
+
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_NAME);
+    HazelcastClientWrapper hazelcastClientWrapper = new HazelcastClientWrapper(settings);
+    hazelcastClientWrapper.stop();
+  }
+
+  @Test
+  public void missing_CLUSTER_LOCALENDPOINT_throw_ISE() {
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("LocalEndPoint have not been set");
+
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_LOCALENDPOINT);
+    HazelcastClientWrapper hazelcastClientWrapper = new HazelcastClientWrapper(settings);
+    hazelcastClientWrapper.stop();
+  }
+
+  @Test
+  public void client_must_connect_to_hazelcast() {
+    int port = NetworkUtils.freePort();
+    // Launch a fake Hazelcast instance
+    HazelcastInstance hzInstance = HazelcastTestHelper.createHazelcastCluster("client_must_connect_to_hazelcast", port);
+    Settings settings = createClusterSettings("client_must_connect_to_hazelcast", "localhost:" + port);
+    HazelcastClientWrapper hazelcastClientWrapper = new HazelcastClientWrapper(settings);
+    hazelcastClientWrapper.start();
+    assertThat(hazelcastClientWrapper.getConnectedClients()).hasSize(1);
+    assertThat(hazelcastClientWrapper.getClientUUID()).isNotEmpty();
+    hazelcastClientWrapper.stop();
+  }
+
+  @Test
+  public void client_must_be_able_to_retrieve_clustered_objects() {
+    int port = NetworkUtils.freePort();
```

Comment:
I would create a test method for each method to test (`getSet`, `getMap`, ...) and since starting up an Hazelcast cluster is resource and time consuming, I would use a static method annotated with `@BeforeClass` to set it up once for all test methods